src/aws-cpp-sdk-core/source/smithy/client/AwsSmithyClientBase.cpp

```diff
@@ -202,6 +202,21 @@ void AwsSmithyClientBase::AttemptOneRequestAsync(std::shared_ptr<AwsSmithyClient
         return;
     }
 
+    interceptor::InterceptorContext context{};
+    context.SetRequest(pRequestCtx->m_httpRequest);
+    for (const auto& interceptor : m_interceptors)
+    {
+        auto modifiedRequest = interceptor->ModifyRequest(context);
```

Comment:
for the interceptor API,
please rename ModifyRequest into something like ModifyBeforeSigningRequest to be compliant with the internal reference design.

Reply:
yeah that was a miss on my part, [it aligns with rusts](https://github.com/smithy-lang/smithy-rs/blob/3871e9aed7ed6fbecdc45a76c7a5b75f17a941b5/rust-runtime/inlineable/src/http_checksum_required.rs#L29C8-L29C29) as well, updating